### DISCUSSION TITLE
ERROR: No matching distribution found for pillow~=10.4 (from crawl4ai[all])

### DISCUSSION THREAD

**Discussion by wwwmaster1** [unclecode/crawl4ai#514] [Forums - Q&A]
Max version on AWS EC2 Linux is 9.5.0

**Comment by wwwmaster1** [comment]
![image](https://github.com/user-attachments/assets/40098020-75f2-40ac-aa24-ac19cfbe1461)

**Comment by wwwmaster1** [comment]
Sorry, new error: ERROR: Could not find a version that satisfies the requirement tf-playwright-stealth>=1.1.0 (from crawl4ai[all]) (from versions: 0.0.1, 0.0.2, 0.0.3, 0.0.4, 0.0.5, 0.0.6)
ERROR: No matching distribution found for tf-playwright-stealth>=1.1.0 (from crawl4ai[all])

**Comment by wwwmaster1** [comment]
and...

ERROR: Could not find a version that satisfies the requirement numpy<3,>=1.26.0 (from crawl4ai[all]) (from versions: 1.3.0, 1.4.1, 1.5.0, 1.5.1, 1.6.0, 1.6.1, 1.6.2, 1.7.0, 1.7.1, 1.7.2, 1.8.0, 1.8.1, 1.8.2, 1.9.0, 1.9.1, 1.9.2, 1.9.3, 1.10.0.post2, 1.10.1, 1.10.2, 1.10.4, 1.11.0, 1.11.1, 1.11.2, 1.11.3, 1.12.0, 1.12.1, 1.13.0, 1.13.1, 1.13.3, 1.14.0, 1.14.1, 1.14.2, 1.14.3, 1.14.4, 1.14.5, 1.14.6, 1.15.0, 1.15.1, 1.15.2, 1.15.3, 1.15.4, 1.16.0, 1.16.1, 1.16.2, 1.16.3, 1.16.4, 1.16.5, 1.16.6, 1.17.0, 1.17.1, 1.17.2, 1.17.3, 1.17.4, 1.17.5, 1.18.0, 1.18.1, 1.18.2, 1.18.3, 1.18.4, 1.18.5, 1.19.0, 1.19.1, 1.19.2, 1.19.3, 1.19.4, 1.19.5, 1.20.0, 1.20.1, 1.20.2, 1.20.3, 1.21.0, 1.21.1, 1.21.2, 1.21.3, 1.21.4, 1.21.5, 1.21.6)
ERROR: No matching distribution found for numpy<3,>=1.26.0 (from crawl4ai[all])

**Comment by wwwmaster1** [comment]
And 

ERROR: Could not find a version that satisfies the requirement pydantic>=2.10 (from crawl4ai[all]) (from versions: 0.0.1, 0.0.2, 0.0.3, 0.0.4, 0.0.5, 0.0.6, 0.0.7, 0.0.8, 0.1, 0.2, 0.2.1, 0.3, 0.4, 0.5, 0.6, 0.6.1, 0.6.2, 0.6.3, 0.6.4, 0.7, 0.7.1, 0.8, 0.9, 0.9.1, 0.10, 0.11, 0.11.1, 0.11.2, 0.12, 0.12.1, 0.13, 0.13.1, 0.14, 0.15, 0.16, 0.16.1, 0.17, 0.18, 0.18.1, 0.18.2, 0.19, 0.20a1, 0.20, 0.20.1, 0.21, 0.22, 0.23, 0.24, 0.25, 0.26, 0.27a1, 0.27, 0.28, 0.29, 0.30, 0.30.1, 0.31, 0.31.1, 0.32, 0.32.1, 0.32.2, 1.0b1, 1.0b2, 1.0, 1.1, 1.1.1, 1.2, 1.3, 1.4, 1.5, 1.5.1, 1.6, 1.6.1, 1.6.2, 1.7, 1.7.1, 1.7.2, 1.7.3, 1.7.4, 1.8, 1.8.1, 1.8.2, 1.9.0a1, 1.9.0a2, 1.9.0, 1.9.1, 1.9.2, 1.10.0a1, 1.10.0a2, 1.10.0b1, 1.10.0, 1.10.1, 1.10.2, 1.10.3, 1.10.4, 1.10.5, 1.10.6, 1.10.7, 1.10.8, 1.10.9, 1.10.10, 1.10.11, 1.10.12, 1.10.13, 1.10.14, 1.10.15, 1.10.16, 1.10.17, 1.10.18, 1.10.19, 2.0a1, 2.0a2, 2.0a3, 2.0a4, 2.0b1, 2.0b2, 2.0b3, 2.0, 2.0.1, 2.0.2, 2.0.3, 2.1.0, 2.1.1, 2.2.0, 2.2.1, 2.3.0, 2.4.0, 2.4.1, 2.4.2, 2.5.0b1, 2.5.0, 2.5.1, 2.5.2, 2.5.3)
ERROR: No matching distribution found for pydantic>=2.10 (from crawl4ai[all])

**Comment by wwwmaster1** [comment]
And

ERROR: Could not find a version that satisfies the requirement aiofiles>=24.1.0 (from crawl4ai[all]) (from versions: 0.2.1, 0.3.0, 0.3.1, 0.3.2, 0.4.0, 0.5.0, 0.6.0, 0.7.0, 0.8.0, 22.1.0, 23.1.0, 23.2.0, 23.2.1)
ERROR: No matching distribution found for aiofiles>=24.1.0 (from crawl4ai[all])

**Comment by wwwmaster1** [comment]
And

ERROR: Could not find a version that satisfies the requirement python-dotenv~=1.0 (from crawl4ai[all]) (from versions: 0.1.0, 0.1.2, 0.1.3, 0.1.5, 0.2.0, 0.3.0, 0.4.0, 0.5.0, 0.5.1, 0.6.0, 0.6.1, 0.6.2, 0.6.3, 0.6.4, 0.6.5, 0.7.0, 0.7.1, 0.8.0, 0.8.1, 0.8.2, 0.9.0, 0.9.1, 0.10.0, 0.10.1, 0.10.2, 0.10.3, 0.10.4, 0.10.5, 0.11.0, 0.12.0, 0.13.0, 0.14.0, 0.15.0, 0.16.0, 0.17.0, 0.17.1, 0.18.0, 0.19.0, 0.19.1, 0.19.2, 0.20.0, 0.21.0, 0.21.1)
ERROR: No matching distribution found for python-dotenv~=1.0 (from crawl4ai[all])

**Comment by wwwmaster1** [comment]
And 

ERROR: Could not find a version that satisfies the requirement litellm>=1.53.1 (from crawl4ai[all]) (from versions: 0.1.0, 0.1.1, 0.1.2, 0.1.34, 0.1.201, 0.1.202, 0.1.203, 0.1.204, 0.1.205, 0.1.206, 0.1.207, 0.1.208, 0.1.209, 0.1.210, 0.1.211, 0.1.212, 0.1.213, 0.1.214, 0.1.215, 0.1.216, 0.1.217, 0.1.218, 0.1.219, 0.1.220, 0.1.221, 0.1.222, 0.1.223, 0.1.224, 0.1.225, 0.1.226, 0.1.227, 0.1.228, 0.1.229, 0.1.230, 0.1.231, 0.1.232, 0.1.233, 0.1.234, 0.1.235, 0.1.236, 0.1.2291)
ERROR: No matching distribution found for litellm>=1.53.1 (from crawl4ai[all])

**Comment by wwwmaster1** [comment]
And 

ERROR: Could not find a version that satisfies the requirement aiosqlite~=0.20 (from crawl4ai[all]) (from versions: 0.1.0, 0.2.0, 0.2.1, 0.2.2, 0.3.0, 0.4.0, 0.5.0, 0.6.0, 0.7.0, 0.8.0, 0.8.1, 0.9.0, 0.10.0, 0.11.0, 0.12.0, 0.13.0, 0.14.0, 0.14.1, 0.15.0, 0.16.0, 0.16.1, 0.17.0, 0.18.0, 0.19.0)
ERROR: No matching distribution found for aiosqlite~=0.20 (from crawl4ai[all])

**Comment by unclecode** [comment]
@wwwmaster1 Hey, thanks for reporting this. It looks like the installation is failing because Amazon Linux 2 doesn’t have the newer packages required for `pillow`, `numpy`, `pydantic`, `aiofiles`, `python-dotenv`, `litellm`, and `aiosqlite` that `crawl4ai[all]` needs. Would you be able to upgrade your environment, maybe to Amazon Linux 2023 or a newer Python version, so you can install the latest wheels for these packages?

If upgrading your environment isn’t an option, let me know. I can try putting together a custom Dockerfile or a set of instructions tailored to Amazon Linux 2 so that you can run `crawl4ai` without having to manually downgrade all these dependencies. Let me know what works best for you.

**Comment by wwwmaster1** [comment]
Ah bummer. I was hopeful to get it working on EC2 with Python 3.7.16 because it would work alongside my agentic app. I will likely need to upgrade the environment to 2023 anyway, but not sure when. No worries on your end though, you have enough on your plate. Appreciate the innovations :)

**Comment by unclecode** [comment]
@wwwmaster1 To be honest, I think what you are experiencing isn't something rare. There are enough people who have the same thing. And my goal for Crawl4AI is to build the number one library for data extraction and crawling for anyone in any situation. And eventually I will reach that point. So I definitely consider putting 3.7 into my backlog, and I will test it by hopefully this month, and I will definitely update you.

**Comment by lisa348** [comment]
hello, I have the same question,  my python=3.12, and conda install numpy=2.2.3, the lastest version, same error :
(scrapy) wanglisha@ychen-P8000:~/crawl4ai$ pip install -U crawl4ai
Collecting crawl4ai
  Using cached Crawl4AI-0.4.24-py3-none-any.whl (177 kB)
Collecting requests~=2.26
  Using cached requests-2.32.3-py3-none-any.whl (64 kB)
Collecting python-dotenv~=1.0
  Using cached python_dotenv-1.0.1-py3-none-any.whl (19 kB)
Collecting pyOpenSSL>=24.3.0
  Using cached pyOpenSSL-25.0.0-py3-none-any.whl (56 kB)
Collecting litellm>=1.53.1
  Using cached litellm-1.61.9-py3-none-any.whl (6.9 MB)
Collecting aiofiles>=24.1.0
  Using cached aiofiles-24.1.0-py3-none-any.whl (15 kB)
ERROR: Could not find a version that satisfies the requirement numpy<3,>=1.26.0 (from crawl4ai) (from versions: 1.3.0, 1.4.1, 1.5.0, 1.5.1, 1.6.0, 1.6.1, 1.6.2, 1.7.0, 1.7.1, 1.7.2, 1.8.0, 1.8.1, 1.8.2, 1.9.0, 1.9.1, 1.9.2, 1.9.3, 1.10.0.post2, 1.10.1, 1.10.2, 1.10.4, 1.11.0, 1.11.1, 1.11.2, 1.11.3, 1.12.0, 1.12.1, 1.13.0, 1.13.1, 1.13.3, 1.14.0, 1.14.1, 1.14.2, 1.14.3, 1.14.4, 1.14.5, 1.14.6, 1.15.0, 1.15.1, 1.15.2, 1.15.3, 1.15.4, 1.16.0, 1.16.1, 1.16.2, 1.16.3, 1.16.4, 1.16.5, 1.16.6, 1.17.0, 1.17.1, 1.17.2, 1.17.3, 1.17.4, 1.17.5, 1.18.0, 1.18.1, 1.18.2, 1.18.3, 1.18.4, 1.18.5, 1.19.0, 1.19.1, 1.19.2, 1.19.3, 1.19.4, 1.19.5, 1.20.0, 1.20.1, 1.20.2, 1.20.3, 1.21.0, 1.21.1, 1.21.2, 1.21.3, 1.21.4, 1.21.5, 1.21.6, 1.22.0, 1.22.1, 1.22.2, 1.22.3, 1.22.4, 1.23.0, 1.23.1, 1.23.2, 1.23.3, 1.23.4, 1.23.5, 1.24.0, 1.24.1, 1.24.2, 1.24.3, 1.24.4)
ERROR: No matching distribution found for numpy<3,>=1.26.0 (from crawl4ai)


thank you so much
